src/Test/WordPress/ProxyTest.php

```diff
@@ -2,78 +2,143 @@
+        $mockWPVerifyNonce->expects($this->once())->willReturn(true);
+        $this->mockRequestRouter->expects($this->once())->method('route');
+        $mockWPDie = $this->getFunctionMock('CF\WordPress', 'wp_die');
+        $this->proxy->run();
+    }
+
+    public function testIsCloudFlareCSRFTokenValidReturnsTrueForGet()
+    {
+        $this->assertTrue($this->proxy->isCloudFlareCSRFTokenValid('GET', null));
+    }
+
+    public function testCreateRequestGETProxyClient()
+    {
+        $url = 'testproxyurl.com';
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_GET['proxyURL'] = $url;
+        $_GET['proxyURLType'] = 'CLIENT';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
 
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        $this->assertEquals(\CF\API\Client::ENDPOINT.$url, $request->getUrl());
+    }
+
+    public function testCreateRequestGETProxyPlugin()
+    {
+        $url = 'testproxyurl.com';
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_GET['proxyURL'] = $url;
+        $_GET['proxyURLType'] = 'PLUGIN';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
+
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        $this->assertEquals(\CF\API\Plugin::ENDPOINT.$url, $request->getUrl());
+    }
+
+    public function testCreateRequestNonGET()
+    {
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
+
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        // $this->assertEquals($proxyURL, $request->getUrl());
```

Comment:
Rather than trying to mock `file_get_contents()` we can just do this:

1. ```protected $json;```
2.  In the constructor do ```$this->json = json_decode(file_get_contents('php://input'), true);```
3. Create ```public function setJson($json) { $this->json = $json; }
4. Update ```$body = $this->json```
5. In your test mock the response body with ```$this->proxy->setJson($mockJson);```